File: VB/12_LinksAndActions/AddPdfLaunchAction.vb
Imports Spire.Pdf
Imports Spire.Pdf.Annotations
Imports Spire.Pdf.Graphics
Imports Spire.Pdf.Actions

Namespace AddPdfLaunchAction
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument
			Dim doc As New PdfDocument()

			' Add a new page to the document
			Dim page As PdfPageBase = doc.Pages.Add()

			' Create a PdfLaunchAction to launch a file
			Dim launchAction As New PdfLaunchAction("..\..\..\..\..\..\Data\text.txt")

			' Set the text to be displayed on the page
			Dim text As String = "Click here to open file"

			' Set the font for the text
			Dim font As New PdfTrueTypeFont(New Font("Arial", 13.0F))

			' Set the rectangle for the text
			Dim rect As New RectangleF(50, 50, 230, 15)

			' Draw the text on the page using the specified font and rectangle
			page.Canvas.DrawString(text, font, PdfBrushes.ForestGreen, rect)

			' Create a PdfActionAnnotation for the text
			Dim annotation As New PdfActionAnnotation(rect, launchAction)

			' Add the annotation to the page's annotations collection
			TryCast(page, PdfNewPage).Annotations.Add(annotation)

			' Save the document to a file
			Dim result As String = "AddPdfLaunchAction_out.pdf"
			doc.SaveToFile(result)

			' Close the document
			doc.Close()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/12_LinksAndActions/AddPdfLaunchAction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Fields;
using Spire.Pdf.Annotations;
using Spire.Pdf.Graphics;
using Spire.Pdf.Actions;
using Spire.Pdf.General;
namespace AddPdfLaunchAction
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new PDF document 
            PdfDocument doc = new PdfDocument();

            // Add a page to the document
            PdfPageBase page = doc.Pages.Add();

            // Create a PDF Launch Action that will open a text file
            PdfLaunchAction launchAction = new PdfLaunchAction("..\\..\\..\\..\\..\\..\\Data\\text.txt");

            // Create a PDF Action Annotation with the PDF Launch Action
            string text = "Click here to open file";
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Arial", 13f));
            //////////////////Use the following code for netstandard dlls/////////////////////////
            /*
             PdfTrueTypeFont font = new PdfTrueTypeFont("Arial", 13f ,PdfFontStyle.Regular, true);
            */

            RectangleF rect = new RectangleF(50, 50, 230, 15);
            page.Canvas.DrawString(text, font, PdfBrushes.ForestGreen, rect);
            PdfActionAnnotation annotation = new PdfActionAnnotation(rect, launchAction);

            // Add the PDF Action Annotation to the page
            (page as PdfNewPage).Annotations.Add(annotation);

            String result = "AddPdfLaunchAction_out.pdf";

            // Save the document
            doc.SaveToFile(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
